File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/javadoc/JavadocAlignmentSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavadocAlignmentSettings.class)
public interface JavadocAlignmentSettings {

  Optional<Boolean> alignParameterDescriptions();

  Optional<Boolean> alignThrownExceptionDescriptions();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/javadoc/JavadocAlignmentSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc

import com.fasterxml.jackson.annotation.JsonProperty

data class JavadocAlignmentSettings(

    @JsonProperty("Align parameter descriptions")
    val alignParameterDescriptions: Boolean? = null,

    @JsonProperty("Align thrown exception descriptions")
    val alignThrownExceptionDescriptions: Boolean? = null
)
